src/clockpicker.js

```diff
@@ -427,6 +427,30 @@
 		popover.css(styles);
 	};
 
+	// The input can be changed by the user
+	// So before we can use this.hours/this.minutes we must update it
+	ClockPicker.prototype.parseInputValue = function(){
+		var value = this.input.prop('value') || this.options['default'] || '';
+
+		if (value === 'now') {
+			value = new Date(+ new Date() + this.options.fromnow);
+		}
+		if (value instanceof Date) {
+			value = value.getHours() + ':' + value.getMinutes();
+		}
+
+		value = value.split(':');
+
+		// Minutes can have AM/PM that needs to be removed
+		this.hours = + value[0] || 0;
+		this.minutes = + (value[1] + '').replace(/\D/g, '') || 0;
+
+		if (this.twelvehour) {
+			var period = (value[1] + '').replace(/\d+/g, '').toLowerCase();
+			this.amOrPm = this.hours < 12 || period === 'am' ? 'AM' : 'PM';
```

Comment:
@mattkrick You are right fixed the issue in this commit
https://github.com/JordyMoos/clockpicker/commit/6f0ac5f0eb07b5302ebbc4ed7fd5b312c8fae224